### PR TITLE
Add sample rate field to AAC decoder

### LINKED ISSUE #4649: Rhythm Heaven Megamix - gameplay and sound are faster then normal
Tested on latest nightly-1193.
If you play the game the music will become faster then normal and then the gameplay too.

### PR DESCRIPTION
**Pull request by xperia64**
Fixes #4649 

When AAC was added to HLE, it was entirely RE'd from how Pokemon X used it.

Rhythm Heaven Megamix uses AAC audio with a different sample rate than Pokemon, and by comparing HLE to LLE, it would appear that field 'unknown2' of the audio pipe BinaryResponse corresponds to a sample rate enum.

These sample rates were determined by adjusting the value of the 'unknown2' field and measuring how much the game sped up or slowed down compared to the base sample rate of the game's audio files.

I have also verified that this does not break the audio in Pokemon X.

This is WIP, because I have only tested this implementation with the FDK and ffmpeg decoders. I have not even tried to compile the MediaNDK or WMF implementations.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5195)

### CODE REVIEW

File: src/audio_core/hle/decoder.h [hunk 1]
```diff
@@ -24,6 +24,20 @@ enum class DecoderCodec : u16 {
     AAC,
 };
 
+// I'm guessing that this is a u32 (from when it was unknown)
```
Comment (Subv) [inline src/audio_core/hle/decoder.h:27]: Do games use LDR or LDRB to read it? That should tell you what width this field actually has.

Beside that, please attach your name to the TODO if the comment is worded in a personal manner (ie, TODO(xperia64): xxxx), this is so we know whom to ask if we have questions regarding the TODO.

**Comment by xperia64** [comment]
This fix appears to work on WMF from the MSVC CI build, so MediaNDK is the only other platform that needs to be tested.

**Comment by Dragios** [comment]
I leave a note here as it seems related.
image

**Comment by Dragios** [comment]
> Music also plays with DSP HLE but it play it at a faster rate resulting in high pitch music.
Fixes Pokémon Rumble World music as well

**Comment by B3n30** [comment]
Is this still WIP?

**Comment by xperia64** [comment]
> Is this still WIP?

Well I can't test the NDK implementation.

I can confirm that it appears to work with FDK, FFmpeg, and WMF.

File: src/audio_core/hle/decoder.h (hunk 1 above)
Comment (xperia64) [reply src/audio_core/hle/decoder.h]: I updated the comment. I don't know quite how to dig into how the games read fields like this.

**Comment by xperia64** [comment]
Reviewing the code paths, it appears both the NDK and WMF backends retrieve the ADTS data and sample rate in the same way, and seeing as how WMF appear to be working fine, I'm going to call the NDK backend fine as well. If issues come up when Android is released, it can be addressed then.

I would like to merge this PR.

**Pull request merged**